src/main/java/com/alysaa/geyseradmintools/listeners/AdminToolOnJoin.java

```diff
@@ -27,10 +29,21 @@ public void onPlayerJoin(PlayerJoinEvent e) {
         if (!isFloodGatePlayer) {
             return;
         }
-        ItemStack nether = new ItemStack(Material.NETHER_STAR);
-        ItemMeta netherMeta = nether.getItemMeta();
-        netherMeta.setDisplayName(ChatColor.translateAlternateColorCodes('&', "&6Admin Tools"));
-        nether.setItemMeta(netherMeta);
-        player.getInventory().setItem(Gat.plugin.getConfig().getInt("Slot"), nether);
+        if (player.getInventory().contains(starTool)) {
+            return;
+        }
+        // Either we create a copy of the array that is shorter or just use a for loop to only access the itemstacks of the hotbar
+        ItemStack[] wholeInventory = e.getPlayer().getInventory().getContents();
+        boolean success = false;
+        for (int slot = 0; slot < 9; slot++) {
+            if (wholeInventory[slot] == null) {
+                e.getPlayer().getInventory().setItem(slot, starTool);
+                success = true;
+                break;
+            }
+        }
+        if (!success) {
+            e.getPlayer().sendMessage(ChatColor.RED + "Make room in your hotbar for the star tool!");
+        }
```

Comment:
Neat little loop that only goes through the first 8 ItemStacks of the player's inventoy (the hotbar) to only place the item in the hotbar if theres space.

This means that the item won't override and delete and item, and it won't go in the non-hotbar inventory if we used the addItemStack method (or whatever its called)